File: Projects/CSharp.Examples/Person.cs
﻿namespace CSharp.Examples;

public class Person
{
    public string FirstName { get; set; }

    public string LastName { get; set; }

    public Person(string firstName, string lastName)
    {
        this.FirstName = firstName;
        this.LastName = lastName;
    }

    public string GetFullName()
    {
        string fullname = $"{this.LastName}, {this.FirstName}";
        return fullname;
    }
}

File: Projects/FSharp.Examples/Person.fs
namespace Fsharp.Examples

// The CSharpy way
type Person(firstname, lastname) =

    let mutable firstname: string = firstname
    let mutable lastname: string  = lastname

    member _.FirstName with get() = firstname and set(value) = firstname <- value
    member _.LastName with get() = lastname and set(value) = lastname <- value

    member _.GetFullName() = $"{firstname}, {lastname}" 

// F# "normal" way
module Person =
    
    type Person = { FirstName: string; LastName: string }

    let getFullName person = $"{person.FirstName}, {person.LastName}"


